visitors/remove.go

```diff
@@ -0,0 +1,40 @@
+package visitors
+
+import (
+	"github.com/linuxboot/fiano/uefi"
+)
+
+// Remove all firmware files with the given GUID.
+type Remove struct {
+	// Input
+	Predicate func(f *uefi.File) bool
+
+	// Output
+	Matches []*uefi.File
+}
+
+// Visit applies the Remove visitor to any Firmware type.
```

Comment:
Can we actually try to remove by name as well? is it possible to use the Find visitor here?

Reply:
Sounds good! I have to implement the before/after first.